### PR TITLE
feat(HACBS-878): Add decision logic checking Integration pipelines outcome

### PR DESCRIPTION
**Pull request by dirgim**
* Add EnsureApplicationSnapshotPassedAllTests
* Add functions to find all integration tests for a given snapshot
* Adjust the controller to trigger off both build and integration pipelines

### CODE REVIEW

File: controllers/integration/integration_adapter.go [hunk 1]
```diff
@@ -137,8 +222,167 @@ func (a *Adapter) getAllApplicationSnapshots() (*[]appstudioshared.ApplicationSn
...
+			if taskRunResult.Name == "HACBS_TEST_OUTPUT" {
+				var testOutput map[string]interface{}
+				err := json.Unmarshal([]byte(taskRunResult.Value), &testOutput)
+				if err != nil {
+					return false, err
+				}
+				a.logger.Info("Found a task with HACBS_TEST_OUTPUT",
+					"taskRun.Name", taskRun.PipelineTaskName,
+					"taskRun Result", testOutput["result"])
+				if testOutput["result"] != "SUCCESS" && testOutput["result"] != "SKIPPED" {
```
Comment (hongweiliu17) [inline controllers/integration/integration_adapter.go:365]: Should `&&` be changed to `||`?

Comment (hongweiliu17) [reply controllers/integration/integration_adapter.go:365]: Sorry, I am wrong, you are right.

Comment (dirgim) [reply controllers/integration/integration_adapter.go:365]: No need to apologize, the condition is kinda hard to read, I'll see if I can make it a bit more readable.

**Review by jsztuka — APPROVED** [review]
lgtm

**Review by hongweiliu17 — APPROVED** [review]
LGTM

**Review by srivickynesh — APPROVED** [review]
lgtm

File: controllers/integration/integration_adapter.go (hunk 1 above)
Comment (dirgim) [reply controllers/integration/integration_adapter.go]: Changed the condition to be more readable and added an explanatory comment above it

Comment (hongweiliu17) [reply controllers/integration/integration_adapter.go]: Thank you. It is clearer now.

**Pull request merged**